dired-sidebar.el

```diff
@@ -696,8 +704,10 @@ window selection."
                (progn
                  (switch-to-buffer buf-name)
                  (dired-sidebar-update-state (current-buffer)))
-             ;; Copied from `dired-find-file'.
-             (find-file dired-file-name)
+             (if (and dired-sidebar-one-instance (file-directory-p dired-file-name))
+                 (find-alternate-file dired-file-name)
```

Comment:
You're making use of the fact that find-alternate-file kills the dired-sidebar buffer right?

I think it might be more clear -maintenance wise- to use find-file together with kill-buffer.

Reply:
`find-alternate-file` is better, because it will remember the last cursor when switching directories.